### ISSUE TITLE
formatISO number of required arguments. typings vs documentation

### ISSUE THREAD

**Issue by felixsanz** [date-fns/date-fns#2592]
This page says that formatISO requires only 1 argument: https://date-fns.org/v2.23.0/docs/formatISO

But in the real world....

```
Expected at least 2 arguments, but got 1.
(alias) formatISO(dirtyDate: any, dirtyOptions: any, ...args: any[]): string
```

Which one is correct? TS types or docs?

---

### Wrong:

```js
console.log(date.formatISO(date.parse('2016-07-02 00:00', 'yyyy-MM-dd HH:mm', new Date(), {})))
```
### Good:

```js
console.log(date.formatISO(date.parse('2016-07-02 00:00', 'yyyy-MM-dd HH:mm', new Date(), {}), {}))
```

**Comment by fturmel** [comment]
It's one required argument, and both the docs and TS types are correct. The source file for `formatISO` is still JavaScript but it has proper Typescript types defined here: https://github.com/date-fns/date-fns/blob/a3e7e6e20f2f01e98e1e3c9b5d7b54c74a1a2649/typings.d.ts#L452-L458

As far as your test cases go, both are missing closing parenthesis... If you add a closing parenthesis and remove the empty options object (which shouldn't matter but are not necessary), things work just fine for me...

```js
import { parse, formatISO } from "date-fns";

console.log(
  formatISO(parse("2016-07-02 00:00", "yyyy-MM-dd HH:mm", new Date()))
); // 2016-07-02T00:00:00-04:00
```

**Comment by fturmel** [comment]
As I was writing the previous comment, something came to mind. Are you using Deno by any chance?

`formatISO` is still a JavaScript file in the source code and the project's global TS type definitions are not compatible with Deno at this point. The Deno experience with this library is very uneven right now, going from native TypeScript source, to untyped JS, to plain broken (see #2545).

So, for some functions that are still JS sources, you may need to provide all arguments (even optional ones) so the Deno compiler doesn't complain. I believe the goal of `date-fns` is to convert all functions to TypeScript, but the project isn't quite there yet.

As you can see in `formatISO` there is nothing to indicate that the options argument is optional in the function signature...
https://github.com/date-fns/date-fns/blob/a3e7e6e20f2f01e98e1e3c9b5d7b54c74a1a2649/src/formatISO/index.js#L43

...that info is in the JSDocs
https://github.com/date-fns/date-fns/blob/a3e7e6e20f2f01e98e1e3c9b5d7b54c74a1a2649/src/formatISO/index.js#L13-L16

**Comment by felixsanz** [comment]
The parens i forgot is because i added here `console.log()` for completeness, but it's not on my real code, so i forgot the closing one.

Yes, i'm using deno!

**Comment by fturmel** [comment]
Going to close this if you don't mind. `formatISO` has been converted to TypeScript and merged in v2.27, which might help this specific case. It seems like the TypeScript migration is almost done, and we'll see if that other PR to improve Deno support gets merged eventually.
